### PR TITLE
capg: enable compute api for CAPG to build the node images

### PR DESCRIPTION
**Pull request by cpanato**
This PR enables the compute API to be used in the `k8s-staging-cluster-api-gcp` project to build the nightly images for CAPG

Related PRs:
- prow job: https://github.com/kubernetes/test-infra/pull/22041
- image-builder scripts: https://github.com/kubernetes-sigs/image-builder/pull/445

Fixes: https://github.com/kubernetes/k8s.io/issues/1993

## Open questions
 - It is not clear to me is how to use the credentials for this project in the prow job.

/assign @dims @ameukam @spiffxp

### CODE REVIEW

**Comment by ameukam** [comment]
> It is not clear to me is how to use the credentials for this project in the prow job.

@cpanato Good question. We use [workoadIdentity ](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) to ensure the service account used by the job have the right set of permissions.

**Comment by ameukam** [comment]
Ref: https://github.com/kubernetes/k8s.io/issues/1993

File: infra/gcp/ensure-staging-storage.sh
Comment (cpanato) [reply infra/gcp/ensure-staging-storage.sh:320]: thanks for your review!

**Comment by cpanato** [comment]
> > It is not clear to me is how to use the credentials for this project in the prow job.
> 
> @cpanato Good question. We use [workoadIdentity ](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) to ensure the service account used by the job have the right set of permissions.

thanks @ameukam , do we have any PR as an example of how to set up this? I need to make sure that this job https://github.com/kubernetes/test-infra/pull/22041/files receives the right credential

File: infra/gcp/ensure-staging-storage.sh
Comment (spiffxp) [inline infra/gcp/ensure-staging-storage.sh:321]:
```suggestion
    readonly STAGING_PROJECT="k8s-staging-cluster-api-gcp"
    enable_api "${STAGING_PROJECT}" compute.googleapis.com
    ensure_staging_gcb_builder_service_account "cluster-api-gcp" "k8s-infra-prow-build-trusted"
```

You'll need to add a service account file for this, as you did in https://github.com/kubernetes/k8s.io/pull/1620 (just the yaml file).  Then update the job definition to use the service account, as you did in https://github.com/kubernetes/test-infra/pull/20703

Longer term, this is another special case of https://github.com/kubernetes/test-infra/issues/20652#issuecomment-774233051 (gcp-builder per staging project) cropping up again.  We should implement this across the board this release cycle IMO.

**Review by spiffxp — CHANGES_REQUESTED** [review]
This should unblock you.  I am still interested in seeing all of this move back into GCB though.  

@dims @cpanato have you all looked into https://cloud.google.com/build/docs/building/build-vm-images-with-packer at all? My glance of the CI scripts is that they boil down to `packer` calls

https://cloud.google.com/build/docs/building/build-vm-images-with-packer#required_iam_permissions is something we would want scripted, just for this project

https://cloud.google.com/build/docs/building/build-vm-images-with-packer#creating_a_packer_builder_image is something we might want to push to k8s-staging-test-infra or something

File: infra/gcp/ensure-staging-storage.sh
Comment (ameukam) [reply infra/gcp/ensure-staging-storage.sh:321]: @spiffxp Looking at https://www.packer.io/docs/builders/googlecompute, `cluster-api-gcp` will need more permissions to create the VM disks.

Comment (cpanato) [reply infra/gcp/ensure-staging-storage.sh:321]: ahhh!

Comment (cpanato) [reply infra/gcp/ensure-staging-storage.sh:321]: @spiffxp @ameukam thanks forgot that :( added the suggestions and update the job in test-infra: https://github.com/kubernetes/test-infra/pull/22041

Comment (cpanato) [reply infra/gcp/ensure-staging-storage.sh:321]: regarding this comment:
> Longer term, this is another special case of kubernetes/test-infra#20652 (comment) (gcp-builder per staging project) cropping up again. We should implement this across the board this release cycle IMO.

I will work on that for this cycle but will need some initial help

File: infra/gcp/clusters/projects/k8s-infra-prow-build/prow-build/resources/build-serviceaccounts.yaml
Comment (ameukam) [inline infra/gcp/clusters/projects/k8s-infra-prow-build/prow-build/resources/build-serviceaccounts.yaml:20]: @cpanato this Service Account should be in https://github.com/kubernetes/k8s.io/blob/main/infra/gcp/clusters/projects/k8s-infra-prow-build-trusted/prow-build-trusted/resources/build-serviceaccounts.yaml if you want the prowjob running on `k8s-infra-prow-build-trusted`.

Comment (cpanato) [reply infra/gcp/clusters/projects/k8s-infra-prow-build/prow-build/resources/build-serviceaccounts.yaml]: done, thanks!

**Review by spiffxp — APPROVED** [review]
/hold
Remove when ready to deploy. Or I can do this when at keyboard

**Comment by spiffxp** [comment]
OK, looks like I'll have to PR up a fix to get this working
```
$ ./ensure-staging-storage.sh cluster-api-gcp
# ...
Configuring special case for k8s-staging-cluster-api-gcp
Operation "operations/acf.p2-606075400249-63b89821-c81a-42cf-b6d9-4e2fb8d60b10" finished successfully.
ERROR: (gcloud.iam.service-accounts.create) argument NAME: Bad value [gcb-builder-cluster-api-gcp@k8s-staging-cluster-api-gcp.iam.gserviceaccount.com]: Service account name must be between 6 and 30 characters (inclusive), must begin with a lowercase letter, and consist of lowercase alphanumeric characters that can be separated by hyphens.
Usage: gcloud iam service-accounts create NAME [optional flags]
  optional flags may be  --description | --display-name | --help
```

**Comment by spiffxp** [comment]
OK, re-ran with changes from https://github.com/kubernetes/k8s.io/pull/2057

Should be good to go

**Pull request merged**
